front/src/modules/rollingStock/components/RollingStockEditor/RollingStockEditorCurves.tsx

```diff
@@ -91,15 +100,22 @@ export default function RollingStockEditorCurves({
     [selectedTractionMode]
   );
 
+  const extraColumnData = useMemo(
+    () => ({
+      defaultValue: data?.base_power_class ?? '',
+      data: powerRestrictionsClass,
+      updateData: setPowerRestrictionsClass,
+    }),
+    [data, powerRestrictionsClass]
+  );
+
   const [rollingstockParams, setRollingstockParams] =
     useState<RollingStockSelectorParams>(EMPTY_PARAMS);
 
-  /** given a tractionMode and a list of comfort, return empty EffortCurves */
-  const createEmptyCurves = (tractionMode: string, comforts: Comfort[]) => ({
-    curves: comforts.map((comfort) => createEmptyCurve(comfort)),
-    default_curve: { speeds: [0], max_efforts: [0] },
-    is_electric: tractionMode !== THERMAL_TRACTION_IDENTIFIER,
-  });
+  const [powerRestrictionList, setPowerRestrictionList] = useState<string[] | undefined>(
+    powerRestrictionCodes
+  );
```

Comment:
You can remove all the tests that check that powerRestrictionList is defined
```suggestion
  const [powerRestrictionList, setPowerRestrictionList] = useState<string[]>(
    powerRestrictionCodes || []
  );
```

Reply:
done !